zerver/models.py

```diff
@@ -264,14 +291,14 @@ def realm_filters_for_domain(domain):
 def realm_filters_for_domain_memcached(domain):
     filters = []
     for realm_filter in RealmFilter.objects.filter(realm=get_realm(domain)):
-       filters.append((realm_filter.pattern, realm_filter.url_format_string))
+        filters.append((realm_filter.pattern, realm_filter.url_format_string, realm_filter.pk))
 
     return filters
 
 def all_realm_filters():
     filters = defaultdict(list)
     for realm_filter in RealmFilter.objects.all():
-       filters[realm_filter.realm.domain].append((realm_filter.pattern, realm_filter.url_format_string))
+       filters[realm_filter.realm.domain].append((realm_filter.pattern, realm_filter.url_format_string, realm_filter.pk))
```

Comment:
I guess we're missing a space in this for loop too?  Might as well fix that while you're at it.